frontend/src/main.ts

```diff
@@ -60,10 +61,74 @@ function defineCustomElements() {
   customElements.define("svelte-component", SvelteCustomElement);
 }
 
+interface ExtensionModule {
+  onExtensionInit?: (this: ExtensionModule) => void;
+  onExtensionPageLoad?: (this: ExtensionModule) => void;
+}
+
+declare global {
+  interface Window {
```

Comment:
I did this so that js modules could be used to provide only callbacks for the HTML, without needing to use any of the "lifetime" hooks just to make those callbacks accessible to the HTML. So if you want to put a button in the HTML that calls some function defined in the module, without setting something in the global namespace, you'd need to define the `onExtensionPageLoad` hook, assign your custom function to some name in the global namespace, and use that name in the HTML, which seemed like boilerplate that it would be nice to avoid if possible.

Reply:
I've changed the onExtensionPageLoad example to assign the event handler directly. Since I want to avoid adding stuff to the global namespace and HTML inline event handlers